src/dfcx_scrapi/core/test_cases.py

```diff
@@ -433,3 +433,153 @@ def calculate_coverage(self, coverage_type: int, agent_id: str = None):
+            agent_id = self.agent_id
+
+        intents_covered = {}
+        request = types.test_case.CalculateCoverageRequest(
+            agent=agent_id,
+            type_ = "INTENT",
+        )
+
+        client_options = self._set_region(agent_id)
+        client = services.test_cases.TestCasesClient(
+            credentials=self.creds, client_options=client_options
+        )
+
+        response = client.calculate_coverage(request = request)
+        intents_msg = response.intent_coverage.intents
+
+        for msg in intents_msg:
+            intents_covered[msg.intent] = msg.covered
+
+        return intents_covered
+
+    def get_covered_pages(
+        self,
+        agent_id: str = None,
+        ):
+        """
+        Get the page wise test cases coverage information.
+
+        Args:
+            agent_id: The agent to get coverage from. Format:
+            `projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>`.
+        Returns:
+            response: List of tuples having infomration of source page,
+            target page and coverage.
+        """
+
+        if not agent_id:
+            agent_id = self.agent_id
+
+        pages_covered = []
+
+        request = types.test_case.CalculateCoverageRequest(
+            agent=agent_id,
+            type_= "PAGE_TRANSITION",
+            )
+
+        client_options = self._set_region(agent_id)
+        client = services.test_cases.TestCasesClient(
+            credentials=self.creds, client_options=client_options
+        )
+        response =  client.calculate_coverage(request = request)
+        transitions_covered = response.transition_coverage.transitions
+
+        for transition in transitions_covered:
+            source_page = transition.source.page.display_name
+            target_page = transition.target.page.display_name
+            pages_covered.append((source_page, target_page, transition.covered))
+
+        return pages_covered
+
+    def get_covered_flows(
+        self,
+        agent_id: str = None,
+        ):
+        """
+        Get the flow wise test cases coverage information.
+
+        Args:
+            agent_id: The agent to get coverage from. Format:
+            `projects/<Project ID>/locations/<Location ID>/agents/<Agent ID>`.
+        Returns:
+            response: List of tuples having infomration of source flow,
+            target flow and coverage.
+        """
+
+        if not agent_id:
+            agent_id = self.agent_id
+
+        flows_covered = []
+
+        request = types.test_case.CalculateCoverageRequest(
+            agent=agent_id,
+            type_= "PAGE_TRANSITION",
+            )
+
+        client_options = self._set_region(agent_id)
+        client = services.test_cases.TestCasesClient(
+            credentials=self.creds, client_options=client_options
+        )
+        response =  client.calculate_coverage(request = request)
+        transitions_covered = response.transition_coverage.transitions
```

Comment:
@hkhaitan1 instead of reimplementing this code from scratch, you should call the already implemented `calculate_coverage` here with the appropriate `coverage_type` argument of `3`.

```
transitions_covered = self.calculate_coverage(coverage_type=3, agent_id=agent_id)
```

That saves ~13 lines of code.